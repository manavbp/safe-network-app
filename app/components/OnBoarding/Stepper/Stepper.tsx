import React from 'react';
import { withStyles, styled } from '@material-ui/core/styles';
import indigo from '@material-ui/core/colors/indigo';
import Button from '@material-ui/core/Button';
import MobileStepper from '@material-ui/core/MobileStepper';
import {
    ON_BOARDING,
    BASIC_SETTINGS,
    INTRO,
    HOME
} from '$Constants/routes.json';

interface Props {
    showButtons?: boolean;
    noElevation?: boolean;
    theme?: 'default' | 'white';
    onNext: Function;
    onBack: Function;
    steps: number;
    activeStep: number;
    history?: History;
}

const defaultTheme = {
    root: {
        backgroundColor: '#fff'
    },
    dot: {
        backgroundColor: indigo[100]
    },
    dotActive: {
        backgroundColor: indigo[800]
    }
};

const whiteTheme = {
    root: {
        backgroundColor: 'transparent'
    },
    dot: {
        backgroundColor: 'rgba(#fff, 0.3)'
    },
    dotActive: {
        backgroundColor: '#fff'
    }
};

export const Stepper = ( properties ) => {
    const {
        showButtons,
        theme,
        steps,
        activeStep,
        onBack,
        onNext,
        noElevation,
        history
    } = properties;

    const NavButton = styled( Button )( {
        color: indigo[800],
        visibility: showButtons ? 'visible' : 'hidden'
    } );

    const NextButton = (
        <NavButton
            size="small"
            aria-label="OnBoardingNextButton"
            onClick={() => {
                onNext();
                // @ts-ignore
                history.push( BASIC_SETTINGS );
            }}
        >
            Next
        </NavButton>
    );
    const BackButton = (
        <NavButton
            size="small"
            aria-label="OnBoardingBackButton"
            onClick={() => {
                onBack();
                // @ts-ignore
                activeStep === 1
                    ? history.push( ON_BOARDING )
                    : history.push( INTRO );
            }}
        >
            Back
        </NavButton>
    );

    const StyledStepper = withStyles(
        theme === 'white' ? whiteTheme : defaultTheme
    )( MobileStepper );

    return (
        <StyledStepper
            elevation={noElevation ? 0 : 4}
            variant="dots"
            steps={steps}
            position="bottom"
            activeStep={activeStep}
            nextButton={NextButton}
            backButton={BackButton}
        />
    );
};

Stepper.defaultProps = {
    showButtons: true,
    steps: 3,
    theme: 'default',
    activeStep: 0,
    noElevation: false
};
