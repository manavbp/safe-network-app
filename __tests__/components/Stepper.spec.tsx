import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { Stepper } from '$App/components/OnBoarding/Stepper/Stepper';

const shallow = createShallow();

const shallowSetup = ( propOverrides? ) => {
    const props = Object.assign(
        {
            showButtons: true,
            noElevation: false,
            theme: 'default',
            onNext: jest.fn(),
            onBack: jest.fn(),
            history: {
                push: jest.fn()
            },
            steps: 3,
            activeStep: 0
        },
        propOverrides
    );

    const wrapper = shallow( <Stepper {...props} /> );

    return {
        props,
        wrapper
    };
};

describe( 'Stepper', () => {
    it( 'render', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'Go next', () => {
        const { wrapper, props } = shallowSetup();
        wrapper.getElement().props.nextButton.props.onClick();
        expect( props.onNext ).toHaveBeenCalled();
    } );

    it( 'Go back', () => {
        const { wrapper, props } = shallowSetup();
        wrapper.getElement().props.backButton.props.onClick();
        expect( props.onBack ).toHaveBeenCalled();
    } );

    it( 'handle no elevation', () => {
        const { wrapper } = shallowSetup( {
            noElevation: true
        } );
        expect( wrapper.getElement().props.elevation ).toEqual( 0 );
    } );

    it( 'default have elevation', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper.getElement().props.elevation ).not.toEqual( 0 );
    } );
} );
