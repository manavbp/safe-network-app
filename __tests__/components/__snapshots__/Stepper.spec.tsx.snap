// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper render 1`] = `ShallowWrapper {}`;
